subscription/test/search_test.py

```diff
@@ -0,0 +1,108 @@
+import unittest
+import json
+from unittest.mock import patch, MagicMock
+from search import Search
+
+class TestSearch(unittest.TestCase):
+    def setUp(self):
+        self.search = Search()
+        self.test_input = {
+            "CollectionReference": {"EntryTitle": "Sentinel-1 Interferograms (BETA)"},
+            "DataGranule": {
+                "DayNightFlag": "Unspecified",
+                "Identifiers": [
+                    {
+                        "Identifier": "S1-GUNW-A-R-014-tops-20141030_20141018-152945-24166N_21930N-PP-f419-v2_0_3",
+                        "IdentifierType": "ProducerGranuleId"
+                    }
+                ],
+                "ProductionDateTime": "2020-06-25T00:58:15.561Z"
+            },
+            "GranuleUR": "S1-GUNW-A-R-014-tops-20141030_20141018-152945-24166N_21930N-PP-f419-v2_0_3",
+            "MetadataSpecification": {
+                "URL": "https://cdn.earthdata.nasa.gov/umm/granule/v1.6.6",
+                "Name": "UMM-G",
+                "Version": "1.6.6"
+            }
+        }
+        self.search.get_public_search_url = lambda: "https://cmr.earthdata.nasa.gov/search/"
+
+    @patch('search.requests.get')
+    def test_get_concept(self, mock_get):
+        mock_response = MagicMock()
+        mock_response.status_code = 200
+        mock_response.text = json.dumps(self.test_input)
+        mock_get.return_value = mock_response
+
+        self.search.url = 'http://test-url.com'
+        self.search.token = 'test-token'
+
+        result = self.search.get_concept('test-concept-id', '1')
+        self.assertEqual(result, json.dumps(self.test_input))
+
+    def test_get_producer_granule_id(self):
+        result = self.search.get_producer_granule_id(self.test_input)
+        expected = "S1-GUNW-A-R-014-tops-20141030_20141018-152945-24166N_21930N-PP-f419-v2_0_3"
+        self.assertEqual(result, expected)
+
+    @patch('search.os.getenv')
+    def test_get_public_search_url_from_parameter_store(self, mock_getenv):
+        mock_getenv.return_value = 'http://public-search.com/'
```

Comment:
this url exists, having it referenced in a public repository may imply endorsement

Reply:
I changed this value to cmr.earthdata.nasa.gov/search.